cloudinit/config/cc_ansible.py

```diff
@@ -112,18 +121,22 @@ def install(self, pkg_name: str):
             # bootstrap pip if required
             if not which("pip3"):
                 self.distro.install_packages(PIP_PKG)
-            subp(["python3", "-m", "pip", "install", "--user", pkg_name])
+            self.subp(["python3", "-m", "pip", "install", "--user", pkg_name])
 
     def is_installed(self) -> bool:
-        stdout, _ = subp(["python3", "-m", "pip", "list"])
+        stdout, _ = self.subp(["python3", "-m", "pip", "list"])
         return "ansible" in stdout
 
+    def subp(self, command, **kwargs):
+        return subp(args=command, env=self.env, **kwargs)
+
 
 class AnsiblePullDistro(AnsiblePull):
     def __init__(self, distro: Distro):
         self.cmd_pull = ["ansible-pull"]
```

Comment:
Nit: `AnsiblePullDistro` and `AnsiblePullPip`'s `__init__` methods have duplicated set-up, what do you think about abstracting it in the base-class constructor? In that way, it would be more clear what each class does different:

```diff
diff --git a/cloudinit/config/cc_ansible.py b/cloudinit/config/cc_ansible.py
index 0687ec624..d5282bf56 100644
--- a/cloudinit/config/cc_ansible.py
+++ b/cloudinit/config/cc_ansible.py
@@ -67,9 +67,11 @@ CFG_OVERRIDE = "ANSIBLE_CONFIG"
 
 
 class AnsiblePull(abc.ABC):
-    cmd_version: list = []
-    cmd_pull: list = []
-    env: dict = {}
+    def __init__(self, distro: Distro):
+        self.cmd_pull = ["ansible-pull"]
+        self.cmd_version = ["ansible-pull", "--version"]
+        self.distro = distro
+        self.env = os.environ.copy()
 
     def get_version(self) -> Optional[Version]:
         stdout, _ = self.subp(self.cmd_version)
@@ -102,11 +104,8 @@ class AnsiblePull(abc.ABC):
 
 class AnsiblePullPip(AnsiblePull):
     def __init__(self, distro: Distro):
-        self.cmd_pull = ["ansible-pull"]
-        self.cmd_version = ["ansible-pull", "--version"]
-        self.distro = distro
+        super().__init__(distro)
 
-        self.env = os.environ.copy()
         old_path = self.env.get("PATH")
         if old_path:
             self.env["PATH"] = ":".join([old_path, "/root/.local/bin/"])
@@ -132,12 +131,6 @@ class AnsiblePullPip(AnsiblePull):
 
 
 class AnsiblePullDistro(AnsiblePull):
-    def __init__(self, distro: Distro):
-        self.cmd_pull = ["ansible-pull"]
-        self.cmd_version = ["ansible-pull", "--version"]
-        self.distro = distro
-        self.env = os.environ.copy()
-
     def install(self, pkg_name: str):
         if not self.is_installed():
             self.distro.install_packages(pkg_name)
```